org.lflang/src/org/lflang/generator/c/CCompiler.java

```diff
@@ -26,28 +26,34 @@
 package org.lflang.generator.c;
 
 import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
 
 import org.lflang.ErrorReporter;
 import org.lflang.FileConfig;
+import org.lflang.Target;
 import org.lflang.TargetConfig;
+import org.lflang.TargetProperty.Platform;
 import org.lflang.generator.GeneratorBase;
 import org.lflang.generator.GeneratorCommandFactory;
+import org.lflang.generator.GeneratorUtils;
 import org.lflang.generator.LFGeneratorContext;
 import org.lflang.util.FileUtil;
 import org.lflang.util.LFCommand;
 
 /**
- * Responsible for creating and executing the necessary native command to compile code that is generated
- * by the CGenerator. This class invokes a compiler directly.
+ * Responsible for creating and executing the necessary CMake command to compile
+ * code that is generated by the CGenerator. This class uses CMake to compile.
  *
  * @author Soroush Bateni <soroush@utdallas.edu>
- * @author{Edward A. Lee <eal@berkeley.edu>}
- * @author{Marten Lohstroh <marten@berkeley.edu>}
- * @author{Christian Menard <christian.menard@tu-dresden.de}
- * @author{Matt Weber <matt.weber@berkeley.edu>}
+ * @author Edward A. Lee <eal@berkeley.edu>
+ * @author Marten Lohstroh <marten@berkeley.edu>
+ * @author Christian Menard <christian.menard@tu-dresden.de
+ * @author Matt Weber <matt.weber@berkeley.edu>
```

Comment:
Please add yourself :-) That's also appropriate for a bunch of other files here that you've touched with a lot more than just minor refactoring work.